graphs/directed_and_undirected_(weighted)_graph.py

```diff
@@ -226,7 +226,9 @@ def has_cycle(self):
                                 break
                             else:
                                 return True
-                                anticipating_nodes.add(stack[len_stack_minus_one])
+                                # TODO:The following code is unreachable.
```

Comment:
If code is unreachable then determine if the lines are useful.  If they are then move the return statement below them.  If they are not useful then delete them.

Reply:
i tried.. i can't read that part i don't really understand what it's supposed to do